lib/auth/auth.go

```diff
@@ -3962,6 +3976,94 @@ func (a *Server) DeleteKubernetesCluster(ctx context.Context, name string) error
 	return nil
 }
 
+// CreatePlugin creates a new plugin instance.
+func (a *Server) CreatePlugin(ctx context.Context, req *proto.CreatePluginRequest) error {
+	svc := a.pluginExchangeService
```

Comment:
should this be guarded by the same lock used in `SetPluginExchangeService`?

Reply:
Now that I looked at it better, other Enteprise services, e.g. SAMLService, do not use the lock at all:

https://github.com/gravitational/teleport/blob/e3ce1f83c45db68fd5b8c8f633a22d627d337bb1/lib/auth/auth.go#L536-L541

So it might be superfluous here, since we only initialize this once on startup.